src/Nethermind/Nethermind.Core2/Types/ValidatorIndex.cs

```diff
@@ -34,6 +34,8 @@ public ValidatorIndex(ulong number)
         public static ValidatorIndex None => new ValidatorIndex(ulong.MaxValue);
 
         public static implicit operator ulong(ValidatorIndex validatorIndex) => validatorIndex.Number;
+        
+        public static explicit operator int(ValidatorIndex validatorIndex) => (int)validatorIndex.Number;
 
         public static implicit operator ValidatorIndex(ulong value) => new ValidatorIndex(value);
```

Comment:
Suggest that implicit from ValidatorIndex => ulong is okay, but keep the opposite direction explicit. Be consistent across all the types (Slot, Gwei, etc).

To avoid situations like:  ValidatorIndex index = currentEpoch + maxGwei;

Reply:
I see. Can do it.